metadeploy/api/models.py

```diff
@@ -723,11 +736,14 @@ def run(self, ctx, plan, steps, org):
 
 
 class PreflightResultQuerySet(models.QuerySet):
-    def most_recent(self, *, user, plan, is_valid_and_complete=True):
-        kwargs = {"plan": plan, "user": user}
+    def most_recent(
+        self, *, user, plan, is_valid_and_complete=True, scratch_org_id=None
+    ):
+        args = (Q(user=user) | Q(uuid=scratch_org_id),)
+        kwargs = {"plan": plan}
         if is_valid_and_complete:
             kwargs.update({"is_valid": True, "status": PreflightResult.Status.complete})
-        return self.filter(**kwargs).order_by("-created_at").first()
+        return self.filter(*args, **kwargs).distinct().order_by("-created_at").first()
```

Comment:
@wlonk This is still not working for me when not logged in; throwing the same error message:

```
ValueError: 'AnonymousUser' value must be a positive integer or a valid Hashids string.
```

But, I think this entire approach needs a bit more discussion before moving forward. Notably two things:

1. A given Plan may have two preflights -- one for a persistent org if the user is logged in, and another for an MD-created scratch org. Maybe we can punt on this for now, and discuss as part of https://trello.com/c/FK0ioUs7? But we'll need to think through how that will work.

2. Right now, events related to a currently-running preflight/job are sent via a WS channel specific to a user's org_id. That won't work for anonymous users running preflights/jobs on a scratch org.

Reply:
Notes from discussion:

1. We'll punt on this for now.

2. ::
- The `OrgViewSet` will return both info about scratch and persistent orgs, including any pk/id needed to subscribe to that org's WS channel. e.g.

```
{
  <uuid>: {"current_job": <scratch_org_current_job>, "current_preflight": <scratch_org_current_preflight>},
  <org_id>: {"current_job": <persistent_org_current_job>, "current_preflight": <persistent_org_current_preflight>},
}
```

- This endpoint will be fetched by every user, logged in or not, initially on page load.
- The front-end will subscribe to both of these WS channels (`scratch_org: uuid` and `org: org_id`) initially if exists, or on scratch-org creation.